### PR TITLE
qemu_guest_agent: new case of online/offline memory blocks

### PR DESCRIPTION
**Pull request by xiagao**
ID: 1754370
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
depends on: https://github.com/avocado-framework/avocado-vt/pull/2279

**Comment by xiagao** [comment]
(1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_set_mem_blocks.q35: PASS (71.84 s)

**Comment by xiagao** [comment]
@xiangchunfu  Call for your review, thanks.

File: qemu/tests/qemu_guest_agent.py [hunk 1]
```diff
@@ -512,6 +512,131 @@ def gagent_check_set_vcpus(self, test, params, env):
...
+        if mem_blocks[list_index]["online"] is not False:
+            test.fail("%s phys-index memory block is still online"
+                      " via agent." % mem_phys_index)
+
+        error_context.context("Verify the memory block unit size.",
+                              logging.info)
+        mem_size = session.cmd_output(cmd_get_mem)
+        mem_size_aft_offline_guest = mem_size.strip().split()[1]
+        delta = float(mem_size_original) - float(mem_size_aft_offline_guest)
+        if delta != mem_unit_size:
```
Comment (xiangchunfu) [inline qemu/tests/qemu_guest_agent.py]: @xiagao If offline smaller memory size.  maybe delta != mem_unit_size,

Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py]: > @xiagao If offline smaller memory size. maybe delta != mem_unit_size,

I only offline one memory block, and the diff is just the block unit.

**Comment by xiangchunfu** [comment]
Acked by:Xiangchun fu xfu@redhat.com

**Comment by xiagao** [comment]
@vivianQizhu  Could you help review this patch? Many thanks.

File: qemu/tests/qemu_guest_agent.py [hunk 2]
```diff
@@ -512,6 +512,132 @@ def gagent_check_set_vcpus(self, test, params, env):
...
+
+        error_context.context("Offline one memory block which can't be"
+                              " offline.", logging.info)
+        mem_blocks = self.gagent.get_memory_blocks()
+        for memory in mem_blocks:
+            if memory["online"] and memory["can-offline"] is False:
+                mem_obj_index = memory["phys-index"]
+                break
+        else:
+            logging.info("There is no required memory block.")
```
Comment (vivianQizhu) [inline qemu/tests/qemu_guest_agent.py]: "There is no required memory block that can-offline attribute is False."

**Comment by xiagao** [comment]
@vivianQizhu  Updated.
 (1/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_set_mem_blocks.q35: PASS (34.81 s)
 (2/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_set_mem_blocks.q35: PASS (34.22 s)

btw, it depends on: avocado-framework/avocado-vt#2279
Could you review it as well? thanks.

**Comment by vivianQizhu** [comment]
@xiagao Please check my comment: https://github.com/avocado-framework/avocado-vt/pull/2279#discussion_r344013462

**Comment by vivianQizhu** [comment]
Ack.

**Pull request merged**